### PR TITLE
Use the multisig community address for a specific chain

### PR DESCRIPTION
**Pull request by kkosiorowska**
Closes https://github.com/tahowallet/extension/issues/3429

## What
The `gatedParameters` for 0x swaps should be set up for a specific chain. Currently we use `COMMUNITY_MULTISIG_ADDRESS` which is a single hardcoded multisig address. We should look up the community multisig address for a specific chain.

The PR gives us the ability to have the addresses for specific chains. At the moment, the addresses are not yet available. They should be deployed in the coming days. Therefore, by default, we should use the existing address.

https://github.com/tahowallet/extension/assets/23117945/778dce26-b91b-41a5-b77d-0324757e2f91

## To Test
- [ ] Try to swap and check parameters for 0x swaps. By default, the existing `COMMUNITY_MULTISIG_ADDRESS` should be used.

Latest build: [extension-builds-3443](https://github.com/tahowallet/extension/suites/13434138434/artifacts/736437224) (as of Wed, 07 Jun 2023 11:57:09 GMT).

### CODE REVIEW

File: background/constants/index.ts [hunk 1]
```diff
@@ -23,9 +23,12 @@ export const MINUTE = 60 * SECOND
 export const HOUR = 60 * MINUTE
 export const DAY = 24 * HOUR
 
-export const COMMUNITY_MULTISIG_ADDRESS =
+export const DEFAULT_COMMUNITY_MULTISIG_ADDRESS =
   "0x99b36fDbC582D113aF36A21EBa06BFEAb7b9bE12"
 
+// TODO add correct addresses
+export const COMMUNITY_MULTISIG_ADDRESS: Record<string, string> = {}
```
Comment (Shadowfiend) [inline background/constants/index.ts:30]: I think we can be more explicit in what this contains by expanding the type and adjusting the name a bit:

```suggestion
export const COMMUNITY_MULTISIG_ADDRESS_BY_CHAINID: {
  [chainID: string]: string
} = {}
```

File: background/redux-slices/0x-swap.ts [hunk 2]
```diff
@@ -133,11 +134,23 @@ const get0xApiBase = (network: EVMNetwork) => {
   return `${prefix}${base}`
 }
 
-const gatedParameters = {
-  affiliateAddress: COMMUNITY_MULTISIG_ADDRESS,
-  feeRecipient: COMMUNITY_MULTISIG_ADDRESS,
-  buyTokenPercentageFee: SWAP_FEE,
+const getMultisigAddressByNetwork = (network: EVMNetwork) => {
```
Comment (Shadowfiend) [inline background/redux-slices/0x-swap.ts:137]:
```suggestion
function getMultisigAddressByNetwork(network: EVMNetwork) {
```

Let's stick to regular function declarations unless we need the variable-assignment version for specific reasons (e.g. `this` binding, etc). Amongst other things, it gives us better stack traces (as functions attach their names to the stack in a way that `const`s don't).

File: background/redux-slices/0x-swap.ts [hunk 3]
```diff
@@ -133,11 +134,23 @@ const get0xApiBase = (network: EVMNetwork) => {
...
+  // Look up the community multisig address for a specific chain
+  return (
+    COMMUNITY_MULTISIG_ADDRESS[network.chainID] ??
+    DEFAULT_COMMUNITY_MULTISIG_ADDRESS
+  )
 }
+
+const getGatedParameters = (network: EVMNetwork) => {
```
Comment (Shadowfiend) [inline background/redux-slices/0x-swap.ts:145]:
```suggestion
function getGatedParameters(network: EVMNetwork) {
```

**Review by Shadowfiend — COMMENTED** [review]
Quick nonblocking simplification thought.

**Review by Shadowfiend — COMMENTED** [review]
Looks good in principle, will let @hyphenized bring this home since he's feature buddy here.

**Comment by hyphenized** [comment]
This looks good to me, will merge later in case we want to address the non blocking comment.

**Review by hyphenized — APPROVED** [review]

**Pull request merged**